jdtls.ext/com.shengchen.checkstyle.runner/src/main/java/com/shengchen/checkstyle/checker/CheckerService.java

```diff
@@ -83,13 +83,9 @@ public String getVersion() throws Exception {
 
     public Map<String, List<CheckResult>> checkCode(List<String> filesToCheckUris) throws CheckstyleException {
         final List<File> filesToCheck = filesToCheckUris.stream().map(File::new).collect(Collectors.toList());
-        final ICompilationUnit unit = JDTUtils.resolveCompilationUnit(filesToCheck.get(0).toURI());
+        final IFile resource = JDTUtils.findFile(filesToCheck.get(0).toURI().toString());
         try {
-            if (unit != null) {
-                checker.setCharset(unit.getResource().getProject().getDefaultCharset());
-            } else { // File is not in a java project
-                checker.setCharset(JDTUtils.findFile(filesToCheck.get(0).toURI().toString()).getCharset());
-            }
+            checker.setCharset(resource != null ? resource.getCharset() : "utf8");
```

Comment:
Just one more question: Seems that we cannot get the file encoding from the extension client side, is that correct?

Reply:
Yes, refer to https://github.com/microsoft/vscode/issues/824 for encoding-related api, which is still in the backlog.